src/lib/Components/FilterModal.tsx

```diff
@@ -50,42 +36,172 @@ export class FilterModal extends React.Component<ModalProps, State> {
+                <ModalInnerView visible={isComponentMounted}>
+                  <NavigatorIOS
+                    navigationBarHidden={true}
+                    initialRoute={{
+                      component: FilterOptions,
+                      passProps: { closeModal: this.props.closeModal },
                     }}
-                    block
-                    width={100}
-                    variant="secondaryOutline"
-                  >
-                    Ok
-                  </Button>
+                    style={{ flex: 1 }}
+                  />
+                  <Box p={2}>
+                    <Button onPress={() => this.closeModal()} block width={100} variant="secondaryOutline">
+                      Apply
+                    </Button>
+                  </Box>
                 </ModalInnerView>
-              </>
+              </ModalBackgroundView>
             </TouchableWithoutFeedback>
-          </ModalBackgroundView>
-        </TouchableWithoutFeedback>
-      </RNModal>
+          </RNModal>
+        )}
+      </>
     )
   }
 }
+
+interface FilterOptionsState {
+  filterOptions: Array<{ type: string; data: any }>
+}
+
+interface FilterOptionsProps {
+  closeModal: () => void
+  navigator: NavigatorIOS
+}
+export class FilterOptions extends React.Component<FilterOptionsProps, FilterOptionsState> {
+  state = {
+    filterOptions: [],
+  }
+
+  componentDidMount() {
+    const filterOptions = []
+
+    filterOptions.push({
+      type: "Sort by",
+      onTap: this.handleNavigationToSortScreen,
+    })
+
+    this.setState({
+      filterOptions,
+    })
+  }
+
+  handleNavigationToSortScreen = () => {
+    this.props.navigator.push({
+      component: SortOptions,
+    })
+  }
+
+  renderFilterOption = ({ item: { type, onTap } }) => {
+    return (
+      <OptionListItem>
+        <Flex p={2} flexDirection="row" justifyContent="space-between" flexGrow={1}>
+          <Serif size="3">{type}</Serif>
+          <TouchableOptionListItemRow onTouchStart={() => onTap()}>
+            <Serif color={color("black60")} size="3">
+              Default
+            </Serif>
+            <ArrowRightIcon fill="black30" ml={0.3} mt={0.3} />
+          </TouchableOptionListItemRow>
+        </Flex>
+      </OptionListItem>
+    )
+  }
+
+  render() {
+    const { filterOptions } = this.state
+
+    return (
+      <Flex flexGrow={1}>
+        <Flex flexDirection="row" justifyContent="space-between">
+          <Flex alignItems="flex-end" mt={0.5} mb={2}>
+            <CloseIconContainer onTouchStart={() => this.props.closeModal()}>
+              <CloseIcon fill="black100" />
+            </CloseIconContainer>
+          </Flex>
+          <FilterHeader weight="medium" size="4">
+            Filter
+          </FilterHeader>
+          <Sans mr={2} mt={2} size="4">
+            Clear all
+          </Sans>
+        </Flex>
+        <Flex>
+          <FlatList
```

Comment:
The `FlatList` should have a generic type which describes the shape of the items in `data`

e.g. `<FlatList<{type: string, onTap: ...}>`